### PR TITLE
Fix getTotalPaid for calculating precision by order object

### LINKED ISSUE #29124: Function which return total paid for certain order using precision by context instead of order
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When you have active multistore instance then function getTotalPaid in /classes/order/Order.php using current active context  (precision from current currency context). This causes incorrect behaviour and showing error which doesn't really exist.

In multistore context (with TWO decimails in precision)
Screenshot 2022-07-21 at 13 58 58

In multistore context (with ZERO decimails in precision)
Screenshot 2022-07-21 at 13 58 49

### Expected behavior

Correct price validation on order should be calculated by id_currency inside current order object.

### Steps to reproduce

1. Create multistore instance of PrestaShop (CZ and COM, two stores)
2. Set two currencies EUR and CZK
3. For CZK set precision to 0 and for EUR set precision to CZK
4. Set currecy separately for CZ = CZK and COM = EUR
5. Place order for CZK and for EUR currency (EUR currency must have two decimails price on product)
6. Show order from admin area with context of currency with zero precision (CZK)
7. Warning is showing up above payment block

### PrestaShop version(s) where the bug happened

1.7.8.2

### PHP version(s) where the bug happened

7.4.30

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by integritty**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Issue #29124
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29124
| Related PRs       | none
| How to test?      | Issue #29124 
| Possible impacts? | none

### CODE REVIEW

File: classes/order/Order.php [hunk 1]
```diff
@@ -2141,7 +2141,7 @@ public function getTotalPaid($currency = null)
             }
         }
 
-        return Tools::ps_round($total, Context::getContext()->getComputingPrecision());
+        return Tools::ps_round($total, $currency->precision);
```
Comment (matthieu-rolland) [inline classes/order/Order.php:2144]: hi @integritty !

Here you are replacing the computing precision with the display precision. Usually, they are the same value, but it's temporary, at some point the computing precision will always be higher than the computing precision (it's yet to be done)

if `Context::getContext()->getComputingPrecision())` doesn't return the right precision then it's probably what should be fixed...

But maybe there's something I'm not aware of? don't hesitate to explain your fix, thanx 👍

**Review by matthieu-rolland — COMMENTED** [review]
the fix must be reproduced by the QA team before this PR can be accepted, meanwhile I have a feedback about your fix ;)

File: classes/order/Order.php (hunk 1 above)
Comment (matks) [reply classes/order/Order.php:2144]: @integritty friendly reminder

**Comment by matks** [comment]
Hi,

Since we had no news from you for more than 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**